cfgmgr/vlanmgr.cpp

```diff
@@ -0,0 +1,488 @@
+void VlanMgr::processUntaggedVlanMembers(string vlan, const string &members)
+{
+
+    auto consumer_it = m_consumerMap.find(CFG_VLAN_MEMBER_TABLE_NAME);
+    if (consumer_it == m_consumerMap.end())
+    {
+        SWSS_LOG_ERROR("Failed to find tableName:%s", CFG_VLAN_MEMBER_TABLE_NAME);
+        return;
+    }
+    Consumer& consumer = consumer_it->second;
+
+    vector<string> vlanMembers = tokenize(members, ',');
+
+    for (auto vlanMember : vlanMembers)
+    {
+        string member_key = vlan + CONFIGDB_KEY_SEPARATOR + vlanMember;
+
+        /* Directly put it into consumer.m_toSync map */
+        if (consumer.m_toSync.find(member_key) == consumer.m_toSync.end())
+        {
+            vector<FieldValueTuple> fvVector;
+            FieldValueTuple t("tagging_mode", "untagged");
+            fvVector.push_back(t);
+            consumer.m_toSync[member_key] = make_tuple(member_key, SET_COMMAND, fvVector);
+            SWSS_LOG_DEBUG("%s", (dumpTuple(consumer, consumer.m_toSync[member_key])).c_str());
+        }
+        /*
+         * There is pending task from consumer pipe, in this case just skip it.
+         */
+        else
+        {
+            SWSS_LOG_WARN("Duplicate key %s found in table:%s", member_key.c_str(), CFG_VLAN_MEMBER_TABLE_NAME);
+            continue;
+        }
+    }
+
+    doTask(consumer);
+    return;
+}
+
+void VlanMgr::doVlanMemberTask(Consumer &consumer)
+{
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        auto &t = it->second;
+
+        string key = kfvKey(t);
+
+        /* Ensure the key starts with "Vlan" otherwise ignore */
+        if (strncmp(key.c_str(), VLAN_PREFIX, 4))
+        {
+            SWSS_LOG_ERROR("Invalid key format. No 'Vlan' prefix: %s", key.c_str());
+            it = consumer.m_toSync.erase(it);
+            continue;
+        }
+
+        key = key.substr(4);
+        size_t found = key.find(CONFIGDB_KEY_SEPARATOR);
+        int vlan_id;
+        string vlan_alias, port_alias;
+        if (found != string::npos)
+        {
+            vlan_id = stoi(key.substr(0, found));
+            port_alias = key.substr(found+1);
+        }
+        else
+        {
+            SWSS_LOG_ERROR("Invalid key format. No member port is presented: %s",
+                           kfvKey(t).c_str());
+            it = consumer.m_toSync.erase(it);
+            continue;
+        }
+
+        vlan_alias = VLAN_PREFIX + to_string(vlan_id);
+        string op = kfvOp(t);
+
+       // TODO:  store port/lag/VLAN data in local data structure and perform more validations.
+        if (op == SET_COMMAND)
+        {
+            /* Don't proceed if member port/lag is not ready yet */
+            if (!isMemberStateOk(port_alias) || !isVlanStateOk(vlan_alias))
+            {
+                SWSS_LOG_DEBUG("%s not ready, delaying", kfvKey(t).c_str());
+                it++;
+                continue;
+            }
+            string tagging_mode = "untagged";
```

Comment:
can you also update the VLAN_MEMBER scheme in https://github.com/Azure/SONiC/wiki/Configuration ?

Reply:
ok